federation/rest/node.go

```diff
@@ -0,0 +1,72 @@
+package rest
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	"github.com/cortezaproject/corteza-server/federation/rest/request"
+	"github.com/cortezaproject/corteza-server/federation/service"
+	"github.com/cortezaproject/corteza-server/federation/types"
+	"github.com/cortezaproject/corteza-server/federation/util"
+)
+
+type (
+	Node struct {
+		svcNode service.NodeService
+	}
+)
+
+var (
+	ErrInvalidNodeCreateParams = errors.New("create node: missing or invalid parameters")
+	ErrorIdentityMissingToken  = errors.New("identity: token missing")
+)
+
+func (Node) New() *Node {
+	return &Node{
+		svcNode: service.DefaultNode,
+	}
+}
+
+func (ctrl Node) Create(ctx context.Context, r *request.NodeCreate) (interface{}, error) {
+	if r.NodeURI == "" && (r.Domain == "" || r.Name == "") {
+		return nil, ErrInvalidNodeCreateParams
+	}
+
+	n := &types.Node{}
+	if r.NodeURI != "" {
```

Comment:
why are we doing this (biz logic) here? 
controllers should only translate payloads into internal structures and hand them off to service.

decision model: if code will need to be duplicated for CLI/gRPC/graphql/***... it does not belong here...

Reply:
This is what I was trying to achieve, by converting some input into some standard structure. It didn't make sense to create another intermediate structure to store unified data that is later converted to a Node in the service.

Reply:
but pairing URI is not something that only comes in via HTTP proto, correct?